### PR TITLE
Automation script for build process.

### PR DESCRIPTION
**Pull request by jwittner**
Overview
---

Adds metadata and a build script to package up the repo into NuGet and Unity packages.

### CODE REVIEW

File: Assets/MixedRealityToolkit-Examples/MixedReality.Toolkit.Examples.nuspec [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <owners>Microsoft,MixedReality</owners>
+    <licenseUrl>https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/master/License.txt</licenseUrl>
+    <projectUrl>https://github.com/Microsoft/MixedRealityToolkit-Unity</projectUrl>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <description>Examples of the Mixed Reality Toolkit</description>
+    <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+    <releaseNotes>$releaseNotes$</releaseNotes>
+    <tags>Unity MixedReality</tags>
+    <dependencies>
+      <dependency id="Microsoft.MixedReality.Toolkit.SDK" version="$version$" />
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit-Examples/MixedReality.Toolkit.Examples.nuspec]: also depends on Microsoft.MixedReality.Toolkit (or do dependencies chain?)

File: Assets/MixedRealityToolkit/MixedReality.Toolkit.nuspec [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <licenseUrl>https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/master/License.txt</licenseUrl>
+    <projectUrl>https://github.com/Microsoft/MixedRealityToolkit-Unity</projectUrl>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <description>The Mixed Reality Toolkit is a collection of scripts and components intended to accelerate development of applications targeting Microsoft HoloLens and Windows Mixed Reality headsets.</description>
+    <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+    <releaseNotes>$releaseNotes$</releaseNotes>
+    <tags>Unity MixedReality</tags>
+  </metadata>
+  <files>
+        <file src="**" target="" exclude="Sharing\**;Sharing.meta" />
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/MixedReality.Toolkit.nuspec]: we can probably remove this as there is nothing of consequence here to filter out

File: Assets/MixedRealityToolkit/Sharing/MixedReality.Toolkit.Sharing.nuspec [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/Sharing/MixedReality.Toolkit.Sharing.nuspec]: Let's remove this file. The core toolkit will be distributed as a single package.

File: Assets/MixedRealityToolkit-Examples/MixedReality.Toolkit.Examples.nuspec (hunk 1 above)
Comment (jwittner) [reply Assets/MixedRealityToolkit-Examples/MixedReality.Toolkit.Examples.nuspec]: Dependencies do chain. If you'd like I can add it here for explicitness.

File: Assets/MixedRealityToolkit/MixedReality.Toolkit.nuspec (hunk 2 above)
Comment (jwittner) [reply Assets/MixedRealityToolkit/MixedReality.Toolkit.nuspec]: Ah, I see. We ran into an issue in the past where RakNet was not passing customer rules around OSS for leveraging the MRTK so we've been building without it locally for a while now. Is RakNet coming back in? What's the plan for the Sharing stuff?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/MixedReality.Toolkit.nuspec]: We're not sure yet. Let's add it in when we cross that bridge.

**Comment by jwittner** [comment]
@davidkline-ms Pushed updates for all but the dependency chain comment above. Updated the script to improve some error cases and to keep logs for all invocations of Unity along with dropping the 'Sharing' checks.

File: Assets/MixedRealityToolkit-Examples/MixedReality.Toolkit.Examples.nuspec (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit-Examples/MixedReality.Toolkit.Examples.nuspec]: No need. Thanks!

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit-Examples/MixedReality.Toolkit.Examples.nuspec]: No need!

**Comment by StephenHodgson** [comment]
> Although we REALLY need the automation tests in GitHub which will kick off this build automatically once tests are complete.

Agreed

**Pull request merged**